Release v1.3.0 (#207)
airbrake/pybrake: changelog.md

# Changelog
All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [1.3.0] - 2022-05-26
### Added
- Changelog file
- Pyramid middleware
- CherryPy middleware
- FastAPI middleware

## [1.2.0] - 2022-05-11
### Added
- Bottle middleware
- Bottle sample example

### Fixed
- Errors filtered out when middleware involve in backtrace [#191](https://github.com/airbrake/pybrake/issues/191)

## [1.1.0] - 2022-05-09
### Changed
- Package Restructure
- Add Django sample example
- Update Flask sample example

### Fixed
- Fix query stats send to the Airbrake

## [1.0.6] - 2022-04-13
### Changed
- Improve pytest

### Fixed

> - [#183](https://github.com/airbrake/pybrake/issues/183) on Flask request filter

Reviewer: This looks confusing and not useful if read this changelog anywhere else but GitHub. Let's describe #183 here.

  object [#132](https://github.com/airbrake/pybrake/issues/132)
- blocklist_filter attempts to change Immutable [#133](https://githubcom/airbrake/pybrake/issues/133)

## [1.0.4] - 2021-06-07
### Changed
- notifier: make sure performance_stats are always True

### Fixed
- remote_settings: don't overwrite hosts with None:- When `apm_host` or `error_host` is undefined, it'll overwrite whatever value is in the current remote config. We don't want that to happen.
- .pylintrc: disable 'consider-using-with':- This is just noise at the time of writing this commit, but we should probably address it in the future.
- Pybrake query notifier throws error when query is a psycopg2 Composed 
  object [#156](https://github.com/airbrake/pybrake/issues/156)
- Fixes exception for unhashable type: 'Composed' [#157](https://github.com/airbrake/pybrake/pull/157)

## [1.0.3] - 2021-04-13
### Fixed
- Flask Middleware :- Return response when performance_stats is false

## [1.0.2] - 2021-02-16
### Removed
- remote_settings: delete print that's left from debugging

## [1.0.1] - 2021-02-16
### Changed
- Change naming style for remote_settings, settings_data

### Fixed

> - remote_settings: fix fetching config route (The config route setting has no effect due to a bug where pybrake doesn't read that value prior to making a GET request to the notifier config server. The fix is to call `config_route()` on every notifier config GET call. Additionally, we are adding a logic that prevents remote settings to crash the background thread in case `config_route` is a bad value (HTTP lib would raise an error). When an error happens such as 403 Forbidden, we make a 2nd request to the old config route, which was known to work.)

Reviewer: Let's keep formatting at 80 chars per line. This way people can use different editors to read the changelog.

## [1.0.0] - 2021-01-21
### Changed
- Update year to match in every README example.

> - Reluctantly stop making fun of Brits only, since most of the world

Reviewer: 🤯⁇

  writes dates in a strange way.

### Fixed
- Fix typos in recent README changes.
- Update outdated unreleased diff link.

> [Unreleased]: https://github.com//airbrake/pybrake/compare/v1.3.0...HEAD

Reviewer: ```suggestion
[Unreleased]: https://github.com/airbrake/pybrake/compare/v1.3.0...HEAD
```

> [1.3.0]: https://github.com/olivierlacan/keep-a-changelog/compare/v1.2.0...v1.3.0

Reviewer: Broken links

# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Unreleased]

## [1.3.0] - 2022-05-26

### Added

- Changelog file
- Pyramid middleware
- CherryPy middleware
- FastAPI middleware

## [1.2.0] - 2022-05-11

### Added

- Bottle middleware
- Bottle sample example

### Fixed

> - Errors filtered out when middleware involve in backtrace [#191](https://github.com/airbrake/pybrake/issues/191)

Reviewer: ```suggestion
- Errors are filtered out when middleware is involved in backtrace [#191](https://github.com/airbrake/pybrake/issues/191)
```

  of writing this commit, but we should probably address it in the future.
- Pybrake query notifier throws error when query is a psycopg2 Composed
  object [#156](https://github.com/airbrake/pybrake/issues/156)
- Fixes exception for unhashable type: 'Composed' [#157](https://github.com/airbrake/pybrake/pull/157)

## [1.0.3] - 2021-04-13

### Fixed

- Flask Middleware :- Return response when performance_stats is false

## [1.0.2] - 2021-02-16

### Removed

- remote_settings: delete print that's left from debugging

## [1.0.1] - 2021-02-16

### Changed

- Change naming style for remote_settings, settings_data

### Fixed

- remote_settings: fix fetching config route (The config route setting has no 
effect due to a bug where pybrake doesn't read that value prior to making a 
GET request to the notifier config server. The fix is to call `config_route()`
on every notifier config GET call. Additionally, we are adding a logic that 
prevents remote settings to crash the background thread in case `config_route` 
is a bad value (HTTP lib would raise an error). When an error happens such as
403 Forbidden, we make a 2nd request to the old config route, which was known

> to work.)

Reviewer: ```suggestion
to work)
```

## [1.0.0] - 2021-01-21

### Added

- Pybrake SDK

> - Middleware for Flask, Django, celery, aiohttop frameworks

Reviewer: ```suggestion
- Middleware for Flask, Django, celery, aiohttp frameworks
```
